pkg/container/non_root_container_check.go

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package container
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/tractusx"
+)
+
+// validateUserRegex is used to match valid username/uid, group-name/gid
+const validateUserRegex = `(^(\d|[1-9]\d{1,3}|[1-5]\d{4}|6[0-4]\d{3}|65[0-4]\d{2}|655[0-2]\d|6553[0-6])$)|(^[a-z_][a-z0-9_-]*[$]?$)`
+
+type NonRootContainer struct{}
+
+// NewNonRootContainer returns a new check based on tractusx.QualityGuideline interface.
+func NewNonRootContainer() tractusx.QualityGuideline {
+	return &NonRootContainer{}
+}
+
+func (n NonRootContainer) Name() string {
+	return "TRG 4.03 - Non-root container"
+}
+
+func (n NonRootContainer) Description() string {
+	return "Container images shall not run as root for security reasons."
+}
+
+func (n NonRootContainer) ExternalDescription() string {
+	return "https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-03"
+}
+
+func (n NonRootContainer) Test() *tractusx.QualityResult {
+	checkPassed := true
+	var errorDescription string
+	dockerfiles := findDockerfilesAt("./")
+
+	for _, dockerfilePath := range dockerfiles {
+		file, err := dockerfileFromPath(dockerfilePath)
+		if err != nil {
+			fmt.Printf("Could not read Dockerfile from path %s\n", dockerfilePath)
+		}
+
+		if !validateUser(file.user()) {
+			checkPassed = false
+			errorDescription = "Invalid user specified in Dockerfile"
```

Comment:
I think it would be helpful to specify in the error msg which Dockerfile has issue. If there are more Dockerfiles it's not clear which to fix.

Reply:
added the `dockerfilePath` to `errorDescription` in commit 4ea7727